src/tests.cpp

```diff
@@ -118,14 +118,16 @@ void rr(string good_examples_file, string bad_examples_file) {
 
     for (unsigned int i = 1; i <= recur; i++) {
         for (unsigned int q = 0; q < in_queue_cnt; q++) {
-            wl.add_spec(TimedSpec(Comp(Indiv(metric_t::CENQ, q), op_t::GE, i * rate),
+            Comp* spec = new Comp(Indiv(metric_t::CENQ, q), op_t::GE, i * rate);
+            wl.add_spec(TimedSpec(spec,
                                   time_range_t(i * period - 1, i * period - 1),
                                   total_time));
         }
     }
 
+    Comp* spec1 = new Comp(Indiv(metric_t::CENQ, queue1), op_t::GT, Indiv(metric_t::CENQ, queue2));
     wl.add_spec(
-        TimedSpec(Comp(Indiv(metric_t::CENQ, queue1), op_t::GT, Indiv(metric_t::CENQ, queue2)),
+        TimedSpec(spec1,
```

Comment:
Just add a new, no need for a separate variable.

Reply:
[Tests handle pointers more elegantly](https://github.com/all-things-networking/fperf/pull/27/commits/b9bc7fb16bff8b1a55e3f11963048e071ca1ad0b)